crates/sui-bridge/src/sui_client.rs

```diff
@@ -93,100 +93,26 @@ where
     }
 
     /// Query emitted Events that are defined in the given Move Module.
-    /// It paginates results in **transaction granularity** for easier
-    /// downstream processing. Unlike the native event query
-    /// that uses `EventID` as cursor, here the cursor is `TransactionDigest`
-    /// such that we can be sure the last event in the page must be the
-    /// last Bridge event in that transaction.
-    /// Consideration: a Sui transaction block should not contain too many
-    /// Bridge events.
     pub async fn query_events_by_module(
         &self,
         package: ObjectID,
         module: Identifier,
-        // Before we support query by checkpoint, we use `tx_digest`` as cursor.
-        // Note the cursor is exclusive, so a callsite passing tx A as cursor
-        // meaning it is intersted in events in transactions after A, globally ordering wise.
-        cursor: TransactionDigest,
-    ) -> BridgeResult<Page<SuiEvent, TransactionDigest>> {
+        // cursor is exclusive
+        cursor: EventID,
+    ) -> BridgeResult<Page<SuiEvent, EventID>> {
```

Comment:
meat 1 of this PR:
instead of juggling to adapt `TransactionDigest` as the cursor, we simply use the existing RPC `query_events` to do the work.